library/lib/platforms/glfw/glfw_input.cpp

```diff
@@ -110,4 +112,39 @@ void GLFWInputManager::updateControllerState(ControllerState* state)
     }
 }
 
+void GLFWInputManager::updateTouchState(TouchState* state)
+{
+    // Get gamepad state
+    double x, y;
+    glfwGetCursorPos(this->window, &x, &y);
+
+    state->x = x / Application::windowScale;
+    state->y = y / Application::windowScale;
+
+    int clickState = glfwGetMouseButton(window, GLFW_MOUSE_BUTTON_LEFT);
+    if (clickState == GLFW_PRESS)
+    {
+        if (oldTouch.state == TouchEvent::START || oldTouch.state == TouchEvent::STAY)
```

Comment:
Actually this whole oldTouch logic can be factorized in Application, right now it's duplicated between GLFW and Switch
 
The whole TouchEvent should be moved out of the input manager and automated. The input manager should say that "x fingers are on screen at those positions at the current frame" and Application should automatically deduce the "events" by counting the fingers and seing when a finger was removed or placed on screen compared to previous frame.

The notion of "event" is too high level for a platform driver, and it makes for duplicate logic.

Reply:
Moved that part to base InputManagaer